### PR TITLE
lib: make .toArray padTo arg strict length

### PR DESCRIPTION
**Pull request by sublimator**

### CODE REVIEW

File: test/utils-test.js [hunk 1]
```diff
@@ -63,6 +63,12 @@ describe('BN.js/Utils', function() {
       assert.deepEqual(n.toArray('be', 5), [ 0x00, 0x00, 0x12, 0x34, 0x56 ]);
       assert.deepEqual(n.toArray('le', 5), [ 0x56, 0x34, 0x12, 0x00, 0x00 ]);
     });
+    it('should throw when naturally larger than desired length', function() {
```
Comment (indutny) [inline test/utils-test.js]: Add a newline before this, please

File: lib/bn.js [hunk 2]
```diff
@@ -461,8 +461,11 @@ BN.prototype.toArray = function toArray(endian, padTo) {
     }
   }
 
-  if (padTo) {
-    while (res.length < padTo) {
+  if (length) {
+    if (res.length > length)
```
Comment (indutny) [inline lib/bn.js]: It begs to be an assert!

**Comment by sublimator** [comment]
done & done

**Comment by indutny** [comment]
Landed, thanks!

**Pull request closed without merging**